### PR TITLE
Add the iot credential sample, and one script to generate all the stuff of iot credential provider.

### PR DESCRIPTION
**Pull request by ycyang1229**
*Issue #, if available:*
no
*Description of changes:*
Add the iot credential sample, and one script to generate all the stuff of iot credential provider. The script is based on the kvs tutorial and readme, and I hope this script can save the time of people try the iot credential provider at the first time.  This pr is not necessary, so it depends on you need this or not. You can drop this pr if it is not suitable here.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: samples/Samples.h [hunk 1]
```diff
@@ -40,6 +40,12 @@ extern "C" {
 #define SAMPLE_HASH_TABLE_BUCKET_COUNT  50
 #define SAMPLE_HASH_TABLE_BUCKET_LENGTH 2
 
+#define IOT_CORE_CREDENTIAL_ENDPOINT ((PCHAR) "AWS_IOT_CORE_CREDENTIAL_ENDPOINT")
```
Comment (MushMal) [inline samples/Samples.h:43]: These are generic enough to actually go to a public header in the common sometimes in the future

https://github.com/awslabs/amazon-kinesis-video-streams-producer-c/blob/master/src/include/com/amazonaws/kinesis/video/common/Include.h

**Review by MushMal — APPROVED** [review]
Neat!

File: samples/kvsWebRTCClientMaster.c [hunk 2]
```diff
@@ -20,14 +22,18 @@ INT32 main(INT32 argc, CHAR* argv[])
 
     // do trickleIce by default
     printf("[KVS Master] Using trickleICE by default\n");
-    retStatus =
-        createSampleConfiguration(argc > 1 ? argv[1] : SAMPLE_CHANNEL_NAME, SIGNALING_CHANNEL_ROLE_TYPE_MASTER, TRUE, TRUE, &pSampleConfiguration);
+    if (useIot == FALSE) {
```
Comment (hassanctech) [inline samples/kvsWebRTCClientMaster.c:25]: This will always be FALSE, right?

File: samples/kvsWebRTCClientMaster.c [hunk 3]
```diff
@@ -10,6 +10,8 @@ INT32 main(INT32 argc, CHAR* argv[])
     UINT32 frameSize;
     PSampleConfiguration pSampleConfiguration = NULL;
     SignalingClientMetrics signalingClientMetrics;
+    BOOL useIot = FALSE;
```
Comment (jdelapla) [inline samples/kvsWebRTCClientMaster.c:13]: Let's move this to the same place as the other static values added to this sample in Samples.h

File: samples/Common.c [hunk 4]
```diff
@@ -679,20 +679,32 @@ STATUS lookForSslCert(PSampleConfiguration* ppSampleConfiguration)
     return retStatus;
 }
 
-STATUS createSampleConfiguration(PCHAR channelName, SIGNALING_CHANNEL_ROLE_TYPE roleType, BOOL trickleIce, BOOL useTurn,
+STATUS createSampleConfiguration(PCHAR channelName, SIGNALING_CHANNEL_ROLE_TYPE roleType, BOOL trickleIce, BOOL useTurn, BOOL useIot,
```
Comment (jdelapla) [inline samples/Common.c:682]: going to change useIoT to also use a new statically defined global

**Review by jdelapla — COMMENTED** [review]
Going to merge, and then add the fixes in a follow-up PR

File: samples/Samples.h (hunk 1 above)
Comment (ycyang1229) [reply samples/Samples.h:43]: I noticed this thing, but I am not sure which way the team prefers. So I added it at the sample.h first. I noticed the team modifying this sample, so I left it to them. Good to see you. :)

File: samples/Common.c (hunk 4 above)
Comment (ycyang1229) [reply samples/Common.c:682]: Noticed the team modifying this sample, so left it to the team. thanks.

**Pull request merged**